admin/load.php

```diff
@@ -540,21 +543,29 @@ function perflab_enqueue_modules_page_scripts() {
  */
 function perflab_activate_plugin() {
 	// Do not proceed if plugin query arg is not present.
-	if ( empty( $_GET['plugin'] ) ) {
+	if ( empty( $_GET['plugin_slug'] ) ) {
 		return;
 	}
 
-	// The plugin being activated.
-	$plugin = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
+	$plugin_slug = sanitize_text_field( wp_unslash( $_GET['plugin_slug'] ) );
 
-	check_admin_referer( "perflab_activate_plugin_{$plugin}" );
+	check_admin_referer( "perflab_activate_plugin_{$plugin_slug}" );
 
-	if ( ! current_user_can( 'activate_plugin', $plugin ) ) {
+	$plugins = get_plugins( '/' . $plugin_slug );
```

Comment:
I'm somewhat confused by the `'/' . $plugin_slug` argument. I probably just missed this, but it's getting plugins inside of the specified plugin? Shouldn't this rather be getting plugins that are installed since they are standalone?

Reply:
I was confused by that at first, but it's actually a very smart solution. Since this callback is by definition about a single plugin, in order to find its plugin basename (i.e. plugin slug + main file name), we need to call `get_plugins()`.

Looking for the plugin within only the folder we already know is much more efficient than calling `get_plugins()` for _all_ available plugins and searching through those.

Maybe worth clarifying in a comment:
```suggestion
	// Since we know the plugin folder name, for efficiency we can simply look for the plugin main file there.
	$plugins = get_plugins( '/' . $plugin_slug );
```

Reply:
Ah, I see. So it's acting as if the plugin which is in a directory is actually a single PHP file plugin.

Reply:
WP core use this core pattern many places https://github.com/WordPress/wordpress-develop/blob/f2c78fe642b307acbb77f21abf85a45ef35b1ca4/src/wp-admin/includes/class-plugin-upgrader.php#L541 so we reuse that to find the plugin file.

Reply:
Good to know!